airflow/models.py

```diff
@@ -3978,6 +3983,11 @@ def update_state(self, session=None):
         """
 
         dag = self.get_dag()
+
+        # backfills create subsets of dags with the same name
```

Comment:
I feel this is a bit of a hack and adds unnecessary complexity. I still think we could just "REMOVE" the task instances just in the scheduler.

@mistercrunch what do you think?